parser/src/json/compiler.rs

```diff
@@ -352,11 +353,11 @@ impl Compiler {
                     minimum, maximum
                 )
             })?;
-        Ok(self.lexeme(&rx))
+        Ok(self.lexeme(&rx, false))
     }
 
     fn json_simple_string(&mut self) -> NodeRef {
-        self.lexeme(&format!("\"{}*\"", CHAR_REGEX))
+        self.lexeme("(?s:.*)", true)
```

Comment:
Note that this change is orthogonal to the underlying PR -- @mmoskal if using the `CHAR_REGEX` directly is marginally more performant, I can switch it back.